pkg/gce-pd-csi-driver/controller_test.go

```diff
@@ -1596,12 +1596,12 @@ func (cloud *FakeCloudProviderInsertDiskErr) AddDiskForErr(volKey *meta.Key, err
 	cloud.insertDiskErrors[volKey.String()] = err
 }
 
```

Comment:
It would be nice to add some sort of test that we can actually fetch the PVC correctly. Otherwise we're relying only on the k8s integration tests, and it will be harder to catch a regression.

I'll have to think on how to do that, though. The local api server testing setup I usually use will be hard to use in this context. Maybe you can poke around k/k to see if there's a lightweight mock?

Reply:
True! I wanted to discuss about this, hence marked the PR as draft. 
We could set up a mock client, but I'm not sure how useful/ how often it will be used in this repo. Will check in k/k if I find something.

Reply:
I think something where we can confirm the expected PVC query happens, and the returned name is correctly plumbed into the gcp insert request, would be enough.

Reply:
I'm thinking using [fake package](https://pkg.go.dev/k8s.io/client-go/kubernetes/fake) is a good idea. We can create a client with PVC resource and checked returned hostname. Let me know your thoughts on this.

Reply:
Yup, I think that should do the trick.

Be sure to have tests using this, and also when it's nil (ie the hostname flag is not set)

Reply:
Changes are not tested in staging yet, updated the CL to include tests with fake-client and its imports.